core/invoker/src/main/scala/whisk/core/container/ContainerPool.scala

```diff
@@ -394,6 +401,7 @@ class ContainerPool(
     private val warmNodejsKey = WarmNodeJsActionContainerId
     private val nodejsExec = NodeJS6Exec("", None)
     private val WARM_NODEJS_CONTAINERS = 2
+    private val RM_THRESHOLD = 4
```

Comment:
see line 449. I think this is an arbitrary threshold that avoids creating new warm container if the toBeRemoved queue is >= this threshold.